extensions/shared/use-upgrade-flow/index.js

```diff
@@ -38,13 +41,22 @@ export default function useUpgradeFlow( planSlug, onRedirect = noop ) {
 			checkoutUrl: getUpgradeUrl( { plan, planSlug, postId, postType } ),
 			isAutosaveablePost: editorSelector.isEditedPostAutosaveable(),
 			isDirtyPost: editorSelector.isEditedPostDirty(),
+			planData: plan,
 		};
 	}, [] );
 
 	// Alias. Save post by dispatch.
 	const savePost = dispatch( 'core/editor' ).savePost;
 
 	const goToCheckoutPage = async event => {
+		// If this action is available, the feature is enabled to open the checkout
+		// in a modal rather than redirect the user there, away from the editor.
+		if ( hasAction( HOOK_OPEN_CHECKOUT_MODAL ) ) {
+			event.preventDefault();
+			savePost( event ).then( () => doAction( HOOK_OPEN_CHECKOUT_MODAL, { products: [planData] } ) );
```

Comment:
Perhaps they both can happen simulataneously. But not a blocker for this PR. We'll refine this later.

Reply:
Yes I'd agree with you there. We don't have to wait for the former before we call the latter.

Reply:
Might also need error handling, or using `.finally()` if chained.